ofrak_core/ofrak/core/free_space.py

```diff
@@ -134,16 +138,26 @@ async def allocate_bom(
         for obj, segment in segments_to_allocate:
             vaddr, final_size = 0, 0
             if segment.length > 0:
-                allocs = await self.allocate(
-                    segment.access_perms,
-                    segment.length,
-                    min_fragment_size=segment.length,
-                    alignment=bom.segment_alignment,
+                possible_perms = permission_map[segment.access_perms] or (segment.access_perms,)
+                for candidate_permissions in permission_map[segment.access_perms]:
```

Comment:
Should this be something like:
```suggestion
                possible_perms = permission_map[segment.access_perms] if permission_map is not None else (segment.access_perms,)
                for candidate_permissions in possible_perms:
```